sdk/resourcemanager/Azure.ResourceManager.Core/tests/Scenario/GenericResourceOperationsTests.cs

```diff
@@ -0,0 +1,288 @@
+            ResourceGroup rg = await rgOp.WaitForCompletionAsync();
+            var createOp = await StartCreateGenericAvailabilitySetAsync(rg.Id);
+            GenericResource aset1 = await createOp.WaitForCompletionAsync();
+            var data = new GenericResourceData();
+            data.Location = LocationData.EastUS2;
+            var updateOp = await aset1.StartUpdateAsync(data);
+            GenericResource aset2 = await updateOp.WaitForCompletionAsync();
+            Assert.AreEqual(aset1.Data.Id, aset2.Data.Id);
+            Assert.AreEqual(aset1.Data.Name, aset2.Data.Name);
+            Assert.AreEqual(aset1.Data.Type, aset2.Data.Type);
+            Assert.AreEqual(aset1.Data.Location, aset2.Data.Location);
+            Assert.AreEqual(aset1.Data.Plan, aset2.Data.Plan);
+            Assert.AreEqual(aset1.Data.Properties, aset2.Data.Properties);
+            Assert.AreEqual(aset1.Data.Kind, aset2.Data.Kind);
+            Assert.AreEqual(aset1.Data.ManagedBy, aset2.Data.ManagedBy);
+            Assert.AreEqual(aset1.Data.Sku, aset2.Data.Sku);
+            Assert.AreEqual(aset1.Data.Identity, aset2.Data.Identity);
+            Assert.AreEqual(aset1.Data.Tags, aset2.Data.Tags);
+
+            Assert.ThrowsAsync<ArgumentNullException>(async () =>
+            {
+                var updateOp = await aset1.StartUpdateAsync(null);
+                _ = await updateOp.WaitForCompletionAsync();
+            });
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task AddTag()
+        {
+            ResourceGroup rg = await Client.DefaultSubscription.GetResourceGroups().Construct(LocationData.WestUS2).CreateOrUpdateAsync(Recording.GenerateAssetName("testrg"));
+            GenericResource aset1 = await CreateGenericAvailabilitySetAsync(rg.Id);
+            Assert.AreEqual(0, aset1.Data.Tags.Count);
+            GenericResource aset2 = await aset1.AddTagAsync("key", "value");
+            Assert.AreEqual(1, aset2.Data.Tags.Count);
+            Assert.IsTrue(aset2.Data.Tags.Contains(new KeyValuePair<string, string>("key", "value")));
+            Assert.AreEqual(aset1.Data.Id, aset2.Data.Id);
+            Assert.AreEqual(aset1.Data.Name, aset2.Data.Name);
+            Assert.AreEqual(aset1.Data.Type, aset2.Data.Type);
+            Assert.AreEqual(aset1.Data.Location, aset2.Data.Location);
+            Assert.AreEqual(aset1.Data.Plan, aset2.Data.Plan);
+            Assert.AreEqual(aset1.Data.Properties, aset2.Data.Properties);
+            Assert.AreEqual(aset1.Data.Kind, aset2.Data.Kind);
+            Assert.AreEqual(aset1.Data.ManagedBy, aset2.Data.ManagedBy);
+            Assert.AreEqual(aset1.Data.Sku, aset2.Data.Sku);
+            Assert.AreEqual(aset1.Data.Identity, aset2.Data.Identity);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task StartAddTag()
```

Comment:
I think this is a duplicate of the test already in a file called GenericResourceTests.cs

I think we should rename this file since its really testing GenericResourceOperations, and then add in your additional asserts that aren't present in the other versions of the test

Reply:
Sounds good, I'll do that.